content/guides/collaborations/connect-slack-to-group-collabs/3-scaffold-application-code.md

```diff
@@ -0,0 +1,195 @@
+---
+type: quick-start
+hide_in_page_nav: true
+---
+
+# Scaffold application code
+
+With our Slack and Box applications set up, we're ready to set up the
+application which will listen for incoming slash commands and events from
+Slack.
+
+This application will be split up into three parts:
+
+* Setting up the initial application skeleton and configuration information.
+* Setting up the Slack event and command handlers.
+* Connecting those handlers to the required functions in Box.
+
+This first step will walk through creating the files and information needed to
+start the application.
+
+<Choice option='programming.platform' value='node' color='none'>
+
+* Create a local directory for your application.
+* Create a `package.json` file inside the local directory, open it in your
+ preferred editor, copy / paste the following into it, and save / exit the file.
+
+```javascript
+{
+  "name": "box-slack",
+  "version": "1.0.0",
+  "description": "Box Slack integration",
+  "main": "process.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "node process.js"
+  },
+  "author": "Box",
+  "license": "ISC",
+  "dependencies": {
+    "axios": "^0.19.2",
+    "body-parser": "^1.19.0",
+    "box-node-sdk": "^1.33.0",
+    "express": "^4.17.1"
+  }
+}
+```
+
+* Run `npm init` from the terminal / console to install dependencies.
+* Create two files, `process.js` and `config.js` in the local directory.
+* Open `config.js` and save the following default configuration. 
+
+```javascript
+// Okta application credentials
+const verificationToken = exports.verificationToken = 'TOKEN';
+const botToken = exports.botToken = 'TOKEN';
+```
+
+* There are two values above that will need to be replaced with details from
+ your Slack application. Replace the `TOKEN` strings with the appropriate
+ values.
+  * `verificationToken`: Load up your Slack application configuration page.
+   Within the **Basic Information** page, scroll down to the
+   **App Credentials** section. The **Verification Token** string will be
+   available there.
+  * `botToken`: Within your Slack application, go to the **OAuth & Permissions**
+   page. The **Bot User OAuth Access Token** string is available at the top and
+  was auto-populated once the bot was added to your Slack workspace.
+* Open up the blank `process.js` file, copy and paste the following code, and
+ save the file.
+
+```javascript
+const boxSDK = require('box-node-sdk');
+const config = require('./config.js')
+const express = require('express'); 
+const app = express();
+const axios = require('axios');
+const path = require('path');
+const util = require('util');
+const fs = require('fs');
+
+app.use(express.urlencoded({ extended: true }));
+app.use(express.json());
+
+// INSTANTIATE BOX CLIENT
+
+app.post('/event', (req, res) => {
+  //HANDLE INCOMING EVENTS
+});
+
+const handler = (() => {
```

Comment:
I'm not sure why all these methods needs to be wrapper in a handler. 

It would be simpler not to add all these events, start off with `app.post()` and then add function 1-by-1 as we need them. If we keep every function stateless and pure, then we can just pass the right info down to the next bit of code as we write them. My initial attempt looks like this.

```js
const box = require('box-node-sdk');
const slackConfig = require('./slackConfig.js');
const boxConfig = require('./boxConfig.json');
const express = require('express'); 
const app = express();
const axios = require('axios');

const port = process.env.PORT || 3000;
app.use(express.urlencoded({ extended: true }));
app.use(express.json());

const client = box.getPreconfiguredInstance(boxConfig)
  .getAppAuthClient('enterprise');
                
app.post('/event', (req, res) => {
  if (req.body.token !== slackConfig.verificationToken) {
    res.send('Slack Verification Failed');
  }
  process(req, res);
});

const process = async (req, res)  => {
  if (req.data.type === 'event_callback') {
    handleEventCallback(req, res)
  } else if (data.command === '/boxadd') {
    handleBoxAddCommand(req, res)
  } else {
    res.status(400).send();
  }
}

const handleEventCallback = async (req, res) => {
  const eventType = req.data.event.type;
  const channel = req.data.event.channel;
  const userId = req.data.event.user;

  const user = await getSlackUser(userId);
  await processEvent(user, eventType, channel);
  
  res.send();
}
```